### DISCUSSION TITLE
How to add a DrawingGroup as a MenuItem Icon

### DISCUSSION THREAD

**Discussion by josephnarai** [AvaloniaUI/Avalonia#6706] [Q&A] [answered]
I have a WPF app I'm porting to Avalonia and can't find a way to set the Icon from my DrawingGroup

This is how I was doing it in WPF:

```
MenuItem addDualScaleMenuItem = new()
                {
                    Header = "Dual Scale",
                    Icon = new Image
                    {
                        Source = new DrawingImage((DrawingGroup)Application.Current.Resources["DualScale_16x"]),
                        Height = 16
                    }
                };
```

With the DrawingGroup defined in a Dictionary in App.xaml

```
<DrawingGroup x:Key="DualScale_16x">
                        <GeometryDrawing Brush="#00FFFFFF" Geometry="F1M16,16L0,16 0,0 16,0z" />
                        <GeometryDrawing Brush="White" Geometry="F1M1,8L5,8 5,7 1,7z M1,3L5,3 5,2 1,2z M1,13L5,13 5,12 1,12z" />
                        <GeometryDrawing Brush="White" Geometry="F1M10,8L14,8 14,7 10,7z M10,3L14,3 14,2 10,2z M10,13L14,13 14,12 10,12z" />
                    </DrawingGroup>
```

Avalonia doesn't seem to have any constructors that take a DrawingGroup...

Any advice would be greatly appreciated!

**Comment by MarkusKgit** [comment, marked as answer]
I just tried it and your code almost worked ;-) Just had to set the `Drawing` property of the `DrawingImage` manually

```xaml
MenuItem subItem = new()
{
    Header = "Dual Scale",
    Icon = new Image()
    {
        Source = new DrawingImage()
        {
            Drawing = (DrawingGroup)Application.Current.Resources["DualScale_16x"]
        },
        Height = 16
    }
};
```

But it only works for sub menus, not the top level menu (see [here](https://github.com/AvaloniaUI/Avalonia/issues/4897))

**Reply by josephnarai** [reply]
Thanks again! It's both great and frustrating that Avalonia is close to WPF but in some cases quite different (mostly better once you work out how to implement a feature!)

**Reply by robloo** [reply]
It would be a very easy change to add a new `DrawingImage` constructor that takes a drawing parameter. It sounds like this would match WPF as well. Should a new issue be created to track this and get it added?

**Reply by robloo** [reply]
Created a feature request: https://github.com/AvaloniaUI/Avalonia/issues/6728
